examples/rpc-db/src/main.rs

```diff
@@ -47,26 +38,10 @@ async fn main() -> eyre::Result<()> {
     let spec = Arc::new(ChainSpecBuilder::mainnet().build());
     let factory = ProviderFactory::new(db.clone(), spec.clone());
 
-    // 2. Setup blcokchain tree to be able to receive live notifs
-    // TODO: Make this easier to configure
-    let provider = {
-        let consensus = Arc::new(BeaconConsensus::new(spec.clone()));
-        let exec_factory = ExecutionFactory::new(spec.clone());
-
-        let externals = TreeExternals::new(db.clone(), consensus, exec_factory, spec.clone());
-        let tree_config = BlockchainTreeConfig::default();
-        let (canon_state_notification_sender, _receiver) =
-            tokio::sync::broadcast::channel(tree_config.max_reorg_depth() as usize * 2);
-
-        let tree = ShareableBlockchainTree::new(BlockchainTree::new(
-            externals,
-            canon_state_notification_sender,
-            tree_config,
-            None,
-        )?);
-
-        BlockchainProvider::new(factory, tree)?
-    };
+    // 2. Setup the blockchain provider using only the database provider and a noop for the tree to
+    //    satisfy trait bounds. Tree is not used in this example since we are only operating on the
+    //    disk and don't handle new blocks/live sync etc, which is done by the blockchain tree.
+    let provider = BlockchainProvider::new(factory, NoopBlockchainTree::default())?;
```

Comment:
would go as far as add `BlockchainProvider::offline(factory)` or something, to indicate that there's no live syncing happening